h/static/scripts/controllers.coffee

```diff
@@ -170,45 +198,43 @@ class ViewerController
     loaded = []
 
     loadAnnotations = ->
-      if annotator.tool is 'highlight'
+      if annotationUI.tool is 'highlight'
         return unless auth.user
         query = user: auth.user
 
-      for p in annotator.providers
+      for p in crossframe.providers
         for e in p.entities when e not in loaded
           loaded.push e
-          store.SearchResource.get angular.extend(uri: e, query), (results) ->
-            annotator.loadAnnotations(results.rows)
+          r = store.SearchResource.get angular.extend(uri: e, query), (results) ->
+            annotationMapper.loadAnnotations(results.rows)
 
       streamfilter.resetFilter().addClause('/uri', 'one_of', loaded)
 
-      if auth.user and annotator.tool is 'highlight'
+      if auth.user and annotationUI.tool is 'highlight'
         streamfilter.addClause('/user', auth.user)
 
       streamer.send({filter: streamfilter.getFilter()})
 
-    $scope.$watch (-> annotator.tool), (newVal, oldVal) ->
+    $scope.$watch (-> annotationUI.tool), (newVal, oldVal) ->
       return if newVal is oldVal
       $route.reload()
 
-    $scope.$watchCollection (-> annotator.providers), loadAnnotations
+    $scope.$watchCollection (-> crossframe.providers), loadAnnotations
```

Comment:
As mentioned above, this seems like the wrong place for this. It should be the responsibility of the `AnnotationSync` component to ensure that the right annotations are loaded in all frames.